features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/DocumentEnricher.java

```diff
@@ -151,68 +163,40 @@ private static boolean isPrivateAddress(String ipAddress) {
     }
 
     private Optional<NodeDocument> getNodeInfoFromCache(final String location, final String ipAddress, final ContextKey contextKey, final String value) {
-        final NodeInfoKey key = new NodeInfoKey(location, ipAddress, contextKey, value);
+
+        Optional<NodeDocument> nodeDocument;
+        final NodeMetadataKey metadataKey = new NodeMetadataKey(contextKey, value);
```

Comment:
Let's only perform this if `contextKey != null && !Strings.isNullOrEmpty(value)` and remove that check from the cache loader (i.e. getNodeInfoFromMetadataContext)